File: app/src/main/java/com/juniperphoton/myerlist/util/AppConfig.java
package com.juniperphoton.myerlist.util;


import com.juniperphoton.myerlist.App;

public class AppConfig {
    public static boolean hasLogined() {
        String access_token = LocalSettingUtil.getString(App.getInstance(), Params.ACCESS_TOKEN_KEY);
        return access_token != null;
    }

    public static String getSid() {
        return LocalSettingUtil.getString(App.getInstance(), Params.SID_KEY);
    }

    public static String getAccessToken() {
        return LocalSettingUtil.getString(App.getInstance(), Params.ACCESS_TOKEN_KEY);
    }

    public static boolean isInOfflineMode() {
        return LocalSettingUtil.checkKey(App.getInstance(), Params.OFFLINE_MODE);
    }

    public static boolean addToBottom() {
        if (!LocalSettingUtil.checkKey(App.getInstance(), Params.SETTING_ADD_TO_BOTTOM_KEY)) {
            return true;
        }
        return LocalSettingUtil.getBoolean(App.getInstance(), Params.SETTING_ADD_TO_BOTTOM_KEY, true);
    }
}

File: app/src/main/java/com/juniperphoton/myerlist/util/AppConfig.kt
package com.juniperphoton.myerlist.util


import com.juniperphoton.myerlist.App

object AppConfig {
    val logined: Boolean
        get() = LocalSettingUtil.getString(App.instance!!, Params.ACCESS_TOKEN_KEY) != null

    val sid: String?
        get() = LocalSettingUtil.getString(App.instance!!, Params.SID_KEY)

    val accessToken: String?
        get() = LocalSettingUtil.getString(App.instance!!, Params.ACCESS_TOKEN_KEY)

    val isInOfflineMode: Boolean
        get() = LocalSettingUtil.checkKey(App.instance!!, Params.OFFLINE_MODE)

    fun addToBottom(): Boolean {
        if (!LocalSettingUtil.checkKey(App.instance!!, Params.KEY_ADD_TO_BOTTOM)) {
            return true
        }
        return LocalSettingUtil.getBoolean(App.instance!!, Params.KEY_ADD_TO_BOTTOM, true)
    }
}
